termwiz/src/terminal/windows.rs

```diff
@@ -802,8 +801,8 @@ impl Terminal for WindowsTerminal {
                         "failed to WaitForMultipleObjects: {}",
                         IoError::last_os_error()
                     );
-                } else if result == WAIT_TIMEOUT {
-                    return Ok(None);
+                // } else if result == WAIT_TIMEOUT {
```

Comment:
I don't agree with most of clippy's suggestions.  They're biased to ultimate minimalism, but they cause some information to be lost.  For example, in this case, it is very important for `WAIT_TIMEOUT` to be handled.
In the future, if some work is done here around other cases, the proposed change to implicitly handle it in the common case will be lost and there is risk of regression.